tests/test_utils.py

```diff
@@ -317,7 +317,7 @@ def test_generate_range(self):
         assert generate_range('PHYS_0001', 'PHYS_0005') == ['PHYS_0001', 'PHYS_0002', 'PHYS_0003', 'PHYS_0004', 'PHYS_0005']
         with self.assertRaisesRegex(ValueError, 'could not find numeric part'):
             generate_range('NONUMBER', 'ALSO_NONUMBER')
-        with self.assertRaisesRegex(ValueError, 'evaluates to the same number'):
+        with warns(UserWarning, match='same number'):
             generate_range('PHYS_0001_123', 'PHYS_0010_123')
```

Comment:
Hmm, how does this evaluate on the same page? The page_id here is just `123` for both the start and end?

Reply:
Because the assumption is that the variable number is the last one in the string. For example, Deutsches Textarchiv has the naming convention `<author>_<work>_<year>`, e.g. `kant_aufklaerung_1784` and a possible page ID naming scheme could be `kant_aufklaerung_1784_0001` - in this case it would be wrong to iterate over the `1784` part.

In this case this evaluates to `PHYS_0001_123`, because we take the start id as the basis.